core/src/test/kotlin/io/forestframework/core/http/routing/MediaTypeNegotiatingIntegrationTest.kt

```diff
@@ -0,0 +1,163 @@
+package io.forestframework.core.http.routing
+
+import com.google.inject.Injector
+import io.forestframework.core.Forest
+import io.forestframework.core.config.ConfigProvider
+import io.forestframework.core.http.HttpMethod
+import io.forestframework.core.http.routing.RoutingType.HANDLER
+import io.forestframework.core.modules.WebRequestHandlingModule
+import io.forestframework.ext.api.DefaultStartupContext
+import io.forestframework.ext.api.Extension
+import io.forestframework.ext.api.StartupContext
+import io.forestframework.ext.core.HttpServerExtension
+import io.forestframework.testfixtures.assertStatusCode
+import io.forestframework.testfixtures.get
+import io.forestframework.testfixtures.runBlockingUnit
+import io.forestframework.testsupport.BindFreePortExtension
+import io.vertx.core.Vertx
+import org.junit.jupiter.api.DynamicTest
+import org.junit.jupiter.api.TestFactory
+
+data class ProducesAcceptCase(
+    val description: String,
+    val produces: List<String> = listOf("*/*"),
+    val accept: String = "*/*",
+    val expectedStatusCode: Int = 200,
+    val httpMethod: HttpMethod = HttpMethod.GET
+)
+
+data class ConsumesContentTypeCase(
+    val description: String,
+    val consumes: List<String> = listOf("*/*"),
+    val contentType: String = "*/*",
+    val expectedStatusCode: Int = 200,
+    val httpMethod: HttpMethod = HttpMethod.GET
+)
+
+data class MethodNotAllowedCase(
+    val description: String,
+    val consumes: List<String> = listOf("*/*"),
+    val contentType: String = "*/*",
+    val expectedStatusCode: Int = 200,
+    val httpMethod: HttpMethod = HttpMethod.GET
+)
+
+// Put your test cases here
+val producesAcceptCases = listOf(
+    ProducesAcceptCase("get 200 when matching single", produces = listOf("application/json"), accept = "application/json", expectedStatusCode = 200),
+    ProducesAcceptCase("get 406 when matching single", produces = listOf("application/json"), accept = "text/html", expectedStatusCode = 406),
+
+    ProducesAcceptCase("get 200 when matching multiple", produces = listOf("application/json", "text/plain"), accept = "text/html, application/json;q=0.9, text/plain;q=0.8", expectedStatusCode = 200),
+    ProducesAcceptCase("get 406 when matching multiple", produces = listOf("application/json", "text/plain"), accept = "application/xhtml+xml, text/html;q=0.3", expectedStatusCode = 406),
```

Comment:
`get 406 when matching multiple` or `not matching`?

Reply:
fix it right away!